sql/src/test/scala/org/apache/sedona/sql/functionTestScala.scala

```diff
@@ -459,6 +459,31 @@ class functionTestScala extends TestBaseScala with Matchers with GeometrySample
 
     }
 
+    it("Passed ST_NDims with 2D point") {
+      val test = sparkSession.sql("SELECT ST_NDims(ST_GeomFromWKT('POINT(1 1)'))")
+      assert(test.take(1)(0).get(0).asInstanceOf[Int] == 2)
+    }
+
+    it("Passed ST_NDims with 3D point") {
+      val test = sparkSession.sql("SELECT ST_NDims(ST_GeomFromWKT('POINT(1 1 2)'))")
+      assert(test.take(1)(0).get(0).asInstanceOf[Int] == 3)
+    }
+
+    it("Passed ST_NDims with M coordinates") {
+      val test = sparkSession.sql("SELECT ST_NDims(ST_GeomFromWKT('POINTM(1 1 0.5)'))")
+      assert(test.take(1)(0).get(0).asInstanceOf[Int] == 2)
+    }
+
+    it("Passed ST_NDims with Z coordinates") {
+      val test = sparkSession.sql("SELECT ST_NDims(ST_GeomFromWKT('POINTZ(1 1 0.5)'))")
+      assert(test.take(1)(0).get(0).asInstanceOf[Int] == 3)
+    }
+
+    it("Passed ST_NDims with both Z and M coordinates") {
+      val test = sparkSession.sql("SELECT ST_NDims(ST_GeomFromWKT('POINTZM(1 1 1 0.5)'))")
```

Comment:
According to PostGIS doc (https://postgis.net/docs/ST_NDims.html), XYZM should return 4, but your case returns 3. If you cannot support M, I am ok with this. But please clearly state it in the API doc.

Reply:
Currently the geometry serializer in `sedona-sql` does not support M dimension, 4D geometries with ZM coordinates will have their M coordinates dropped and became 3D geometries. We're working on [a new geometry serializer](https://issues.apache.org/jira/projects/SEDONA/issues/SEDONA-207?filter=allopenissues) to resolve this issue.

Reply:
Thanks, @Kontinuation for clarifying my doubt.

Reply:
@jiayuasu, I made the updates with the test cases and documentation.